exopy/tasks/configs/base_configs.py

```diff
@@ -47,15 +50,15 @@ class BaseTaskConfig(Atom):
 
     def __init__(self, **kwargs):
         super(BaseTaskConfig, self).__init__(**kwargs)
-        # Force check to ensure that the possible default value of task_name
-        # is tested.
-        self.check_parameters()
 
     def check_parameters(self):
-        """The only parameter required is a valid task name.
+        """The only parameter required is a unique task name.
 
         """
-        self.ready = bool(self.task_name)
+        names = []
+        if self.root:
+            names = self.root.get_used_names()
+        self.ready = self.task_name != "" and self.task_name not in names
```

Comment:
To help the user, it would be nice if the UI could provide some feedback (such as a red background for the name). This would require to be able to only validate the name (not hard since you can store the name on the config to make the check cheap).

Reply:
Alternatively you can add a valid_name Bool member.